Xamarin.Forms.Platform.Android/PopupManager.cs

```diff
@@ -112,9 +112,58 @@ void OnActionSheetRequested(Page sender, ActionSheetArguments arguments)
 				builder.Dispose();
 				//to match current functionality of renderer we set cancelable on outside
 				//and return null
+				if (arguments.FlowDirection == FlowDirection.MatchParent)
+				{
+					if ((sender as IVisualElementController).EffectiveFlowDirection.IsRightToLeft())
+					{
+						dialog.SetTitleFlowDirection(FlowDirection.RightToLeft);
+					}
+					else if ((sender as IVisualElementController).EffectiveFlowDirection.IsLeftToRight())
+					{
+						dialog.SetTitleFlowDirection(FlowDirection.LeftToRight);
+					}
+				}
+				else
+				{
+					dialog.SetTitleFlowDirection(arguments.FlowDirection);
+				}
 				dialog.SetCanceledOnTouchOutside(true);
 				dialog.SetCancelEvent((o, e) => arguments.SetResult(null));
 				dialog.Show();
+
+				LayoutDirection layoutDirection = LayoutDirection.Ltr;
+				TextDirection textDirection = TextDirection.Ltr;
+
+				if (arguments.FlowDirection == FlowDirection.LeftToRight)
+				{
+					layoutDirection = LayoutDirection.Ltr;
+					textDirection = TextDirection.Ltr;
+				}
+				else if (arguments.FlowDirection == FlowDirection.RightToLeft)
+				{
+					layoutDirection = LayoutDirection.Rtl;
+					textDirection = TextDirection.Rtl;
+				}
+				else
+				{
+					if ((sender as IVisualElementController).EffectiveFlowDirection.IsRightToLeft())
+					{
+						layoutDirection = LayoutDirection.Rtl;
+						textDirection = TextDirection.Rtl;
+					}
+					else if ((sender as IVisualElementController).EffectiveFlowDirection.IsLeftToRight())
+					{
+						layoutDirection = LayoutDirection.Ltr;
+						textDirection = TextDirection.Ltr;
+					}
+				}
+
+				var listview = dialog.GetListView();
```

Comment:
Same thing here as above

I think you can replace the if statements above by just calling 
`dialog.GetListView().UpdateFlowDirection`